src/fu-self-test.c

```diff
@@ -6010,6 +6028,145 @@ fu_remote_local_func(void)
 	g_assert_true(ret);
 }
 
+static void
+fu_config_migrate_1_9_func(void)
+{
+	const gchar *fake_localconf_fn = "/tmp/fwupd-self-test/var/etc/fwupd/fwupd.conf";
+	const gchar *fake_sysconf_fn = "/tmp/fwupd-self-test/fwupd/fwupd.conf";
+	gboolean ret;
+	g_autofree gchar *localconf_data = NULL;
+	g_autoptr(FuConfig) config = FU_CONFIG(fu_engine_config_new());
+	g_autoptr(GError) error = NULL;
+
+	/* ensure empty tree */
+	fu_self_test_mkroot();
+
+	(void)g_unsetenv("CONFIGURATION_DIRECTORY");
+	(void)g_setenv("FWUPD_SYSCONFDIR", "/tmp/fwupd-self-test", TRUE);
+
+	ret = fu_path_mkdir_parent(fake_sysconf_fn, &error);
+	g_assert_no_error(error);
+	g_assert_true(ret);
+	ret = g_file_set_contents(fake_sysconf_fn,
+				  "# use `man 5 fwupd.conf` for documentation\n"
+				  "[fwupd]\n"
+				  "DisabledPlugins=test;test_ble\n"
+				  "OnlyTrusted=true\n"
+				  "AllowEmulation=false\n",
+				  -1,
+				  &error);
+	g_assert_no_error(error);
+	g_assert_true(ret);
+
+	ret = fu_config_load(config, &error);
+	g_assert_no_error(error);
+	g_assert_true(ret);
+
+	ret = fu_config_set_value(config, "fwupd", "AllowEmulation", "true", &error);
+	g_assert_no_error(error);
+	g_assert_true(ret);
+
+	/* ensure that all keys except AllowEmulation migrated */
+	ret = g_file_get_contents(fake_localconf_fn, &localconf_data, NULL, &error);
+	g_assert_no_error(error);
+	g_assert_true(ret);
+	g_assert_cmpstr(localconf_data,
+			==,
+			"[fwupd]\n"
+			"AllowEmulation=true\n");
+}
+
+static void
+fu_config_set_plugin_defaults(FuConfig *config)
```

Comment:
I guess this works but it's unfortunate we still have everything duplicated both in code and tests.

How about a private function definition we can call for all the defaults in self tests and in code instead.

Reply:
> I guess this works but it's unfortunate we still have everything duplicated

So maybe it's overkill to migrate all the config files? How about just the MSR one? And then it's just one line of duplication.

Reply:
I go back and forth. How about if we record the old md5sums of unmodified and just remove unmodified ones?

Reply:
Oh no, what I mean is maybe we just migrate msr.conf in the tests.

Reply:
Oh like just one?  Would say migrate redfish if just one. That's the one that sploded.m in reports.

Reply:
Okay, I was thinking about this a bit more -- and I do think we want to duplicate the values -- as it actually makes sense to decouple the "what the daemon default should be" and "what the self test expects the value should be" -- for instance, if we change the MSR `MinimumSmeKernelVersion` from `5.18.0` to `5.19.0` in the plugin we want the old default to be ignored too.